### PR TITLE
refactor: remove yq dependency

### LINKED ISSUE #384: drop YQ dependency with bazel-lib 2.0
jq is no longer problematic with the availability of platform binaries such as linux_arm64 rendering use of YQ useless. we should drop use of YQ and replace it with JQ.

### PR DESCRIPTION
**Pull request by thesayyn**
Fixes #384. blocked on bazel-lib 2.0

### CODE REVIEW

File: examples/multi_arch/BUILD.bazel [hunk 1]
```diff
@@ -43,5 +43,5 @@ genrule(
 assert_contains(
     name = "check_digest",
     actual = ":hash",
-    expected = "sha256:7dc510461a627051f9276ec21a26ed9a314e87e94eac1be996fcf48be6cc9aa2",
+    expected = "sha256:a5285d5827015227664ab5a2334a32b9750bd6ec8cfe3b655376b997f020eb43",
```
Comment (thesayyn) [inline examples/multi_arch/BUILD.bazel:46]: made sure that only change is the order of the json keys which leads to digest change.

File: oci/private/image.bzl [hunk 2]
```diff
@@ -84,11 +84,6 @@ If `group/gid` is not specified, the default group and supplementary groups of t
     "_empty_tar": attr.label(default = "empty.tar", allow_single_file = True),
 }
 
-def _format_string_to_string_tuple(kv):
```
Comment (thesayyn) [inline oci/private/image.bzl:87]: unused.

**Comment by rcorre** [comment]
bazel-lib 2.0 is out: https://github.com/aspect-build/bazel-lib/releases/tag/v2.0.0

File: oci/private/tarball.sh.tpl [hunk 3]
```diff
@@ -3,7 +3,7 @@ set -o pipefail -o errexit -o nounset
 
 readonly FORMAT="{{format}}"
 readonly STAGING_DIR=$(mktemp -d)
-readonly YQ="{{yq}}"
+readonly JQ="{{jq_path}}"
```
Comment (thesayyn) [inline oci/private/tarball.sh.tpl:6]: This file gotten out of control. Needs to be tidied up with bsdtar.

**Comment by oh-tarnished** [comment]
Hey all, 

Is this PR ready to be merged ?. Thanks.

**Comment by thesayyn** [comment]
This is now ready to land in 2.x

**Review by alexeagle — APPROVED** [review]

File: oci/private/image_index.sh.tpl [hunk 4]
```diff
@@ -15,17 +15,18 @@ function add_image() {
...
-        platform="${platform}" \
-        manifest="${manifest}" \
-        "${YQ}" --inplace --output-format=json '.manifests += [env(manifest) + {"platform": env(platform)}]' "${output_path}/manifest_list.json"
+        local manifest_blob_path=$("${JQ}" -r '.digest | sub(":"; "/")' <<< ${manifest})
+        local config_blob_path=$("${JQ}" -r '.config.digest | sub(":"; "/")' "${image_path}/blobs/${manifest_blob_path}")
+
+        local platform=$("${JQ}" -c '{"os": .os, "architecture": .architecture, "variant": .variant, "os.version": .["os.version"], "os.features": .["os.features"]} | with_entries(select( .value != null ))' "${image_path}/blobs/${config_blob_path}")
+        "${JQ}" --argjson platform "${platform}" \
+                --argjson manifest "${manifest}" \
+                '.manifests |= [$manifest + {"platform": $platform}]'\
```
Comment (seh) [inline oci/private/image_index.sh.tpl:27]: Per [discussion in the "docker" channel](https://bazelbuild.slack.com/archives/CA3NW13MH/p1718836203461049) of the "Bazel" Slack workspace, this attempt to append to the "manifests" field doesn't work as intended.

Instead, I think we need the following, as suggested in [_jq_'s documentation on _Complex assignments_](https://jqlang.github.io/jq/manual/#complex-assignments):
```shell
'.manifests |= . + [$manifest + {"platform": $platform}]'\
```

Comment (seh) [reply oci/private/image_index.sh.tpl:27]: Please see #633 for a proposed correction.

**Pull request merged**